packages/react-components/src/molecules/__tests__/TabbedCard.test.tsx

```diff
@@ -0,0 +1,144 @@
+    expect(getByText('Second tab group')).toBeVisible();
+  });
+
+  it('cannot switch to a disabled tab', () => {
+    const { getByText, queryByText, getByRole } = render(
+      <TabbedCard
+        {...props}
+        tabs={[firstTab, { ...secondTab, disabled: true }]}
+      />,
+    );
+    expect(getByRole('button', { name: 'First Tab' })).toHaveStyleRule(
+      'font-weight',
+      'bold',
+    );
+
+    fireEvent.click(getByText('Second Tab'));
+    expect(getByRole('button', { name: 'Second Tab' })).toBeDisabled();
+    expect(getByText('Group 1')).toBeVisible();
+    expect(getByText('Group 2')).toBeVisible();
+    expect(queryByText('Second tab group')).not.toBeInTheDocument();
+  });
+
+  it('automatically displays the active tab', () => {
+    const { getByText, queryByText } = render(
+      <TabbedCard {...props} activeTabIndex={1} />,
+    );
+
+    expect(queryByText('Group 1')).not.toBeInTheDocument();
+    expect(queryByText('Group 2')).not.toBeInTheDocument();
+    expect(getByText('Second tab group')).toBeVisible();
+  });
+
+  it('displays tabs as active by default', () => {
+    const { getByRole } = render(<TabbedCard {...props} />);
+    expect(getByRole('button', { name: 'First Tab' })).toBeEnabled();
+    expect(getByRole('button', { name: 'Second Tab' })).toBeEnabled();
+  });
+});
+
+describe('tests the view more functionality', () => {
+  it('display the data once the button is clicked', () => {
+    const { getByText, queryByText } = render(<TabbedCard {...props} />);
+
+    expect(getByText('First Tab')).toBeVisible();
+    expect(getByText('Second Tab')).toBeVisible();
+    expect(getByText('View more interest groups')).toBeVisible();
+
+    expect(getByText('Group 1')).toBeVisible();
+    expect(getByText('Group 2')).toBeVisible();
+    expect(queryByText('Group 3')).not.toBeInTheDocument();
+    fireEvent.click(getByText('View more interest groups'));
+    expect(getByText('Group 3')).toBeVisible();
+    fireEvent.click(getByText('View less interest groups'));
+    expect(queryByText('Group 3')).not.toBeInTheDocument();
+  });
+
+  it('resets on tab change', () => {
+    const { getByText, queryByText } = render(
+      <TabbedCard
+        {...props}
+        tabs={[firstTab, { ...firstTab, tabTitle: 'Second Tab' }]}
+      />,
+    );
+
+    fireEvent.click(getByText('View more interest groups'));
+    expect(getByText('View less interest groups')).toBeVisible();
+
+    fireEvent.click(getByText('Second Tab'));
+    expect(getByText('View more interest groups')).toBeVisible();
+    expect(queryByText('View less interest groups')).not.toBeInTheDocument();
+  });
+
+  it('displays the right number both trimmed and expanded', () => {
+    const { getByText, queryByText, getAllByText } = render(
+      <TabbedCard {...props} />,
+    );
+
+    expect(getByText('Group 1')).toBeVisible();
+    expect(getByText('Group 2')).toBeVisible();
+    expect(getAllByText(/Group/i, { selector: 'div' }).length).toEqual(2);
```

Comment:
```suggestion
    expect(getAllByText(/Group/i, { selector: 'div' }).map(({textContent})=>textContent)).toEqual(['Group 1', 'Group 2']);
```
This will make the test clearer